server/passport.ts

```diff
@@ -129,17 +129,69 @@ function setupPassport(models) {
-            address,
-            chain: registrationChain.id,
-            verification_token: 'MAGIC',
-            verification_token_expires: null,
-            verified: new Date(), // trust addresses from magic
-            last_active: new Date(),
-            user_id: newUser.id,
-            is_magic: true,
-          }, { transaction: t });
+          // create an address on their selected chain
+          let newAddress;
+          if (registrationChain.base === 'substrate') {
+            newAddress = await models.Address.create({
+              address: polkadotAddress,
+              chain: registrationChain.id,
+              verification_token: 'MAGIC',
+              verification_token_expires: null,
+              verified: new Date(), // trust addresses from magic
+              last_active: new Date(),
+              user_id: newUser.id,
+              is_magic: true,
+            }, { transaction: t });
+
+            // if they selected a substrate chain, create an additional address on ethereum
+            // and auto-add them to the eth forum
+            const ethAddressInstance = await models.Address.create({
+              address: ethAddress,
+              chain: 'ethereum',
+              verification_token: 'MAGIC',
+              verification_token_expires: null,
+              verified: new Date(), // trust addresses from magic
+              last_active: new Date(),
+              user_id: newUser.id,
+              is_magic: true,
+            }, { transaction: t });
+
+            await models.Role.create({
+              address_id: ethAddressInstance.id,
+              chain_id: 'ethereum',
+              permission: 'member',
+            });
+          } else {
+            newAddress = await models.Address.create({
+              address: ethAddress,
+              chain: registrationChain.id,
+              verification_token: 'MAGIC',
+              verification_token_expires: null,
+              verified: new Date(), // trust addresses from magic
+              last_active: new Date(),
+              user_id: newUser.id,
+              is_magic: true,
+            }, { transaction: t });
+
+            // if they selected an eth chain, create an additional address on edgeware
+            // and auto-add them to the forum
+            const edgewareAddressInstance = await models.Address.create({
+              address: polkadotAddress,
+              chain: 'edgeware',
+              verification_token: 'MAGIC',
+              verification_token_expires: null,
+              verified: new Date(), // trust addresses from magic
+              last_active: new Date(),
+              user_id: newUser.id,
+              is_magic: true,
+            }, { transaction: t });
+
+            await models.Role.create({
+              address_id: edgewareAddressInstance.id,
+              chain_id: 'edgeware',
+              permission: 'member',
+            }, { transaction: t });
+          }
```

Comment:
if we're doing this, we should do this for the substrate address too (might require removing another place where we auto-create the role)